blockchain/src/blockchain/verify.rs

```diff
@@ -296,4 +301,237 @@ impl Blockchain {
 
         Ok(())
     }
+
+    /// Verifies a `proposed_block`, given the round it is proposed in as well as its valid round if applicable.
+    ///
+    /// For signature verification the signature as well as the supposedly signed payload are needed. They are
+    /// different from the eventual justification as the signature here is a Schnorr signature from the proposer,
+    /// whereas the final proof will be an aggregated bls key of all contributing validators.
+    pub fn verify_macro_block_proposal(
+        &self,
+        proposed_block: MacroBlock,
+        round: u32,
+        valid_round: Option<u32>,
+        signed_data: Vec<u8>,
+        signature: &Signature,
+        signature_only: bool,
+    ) -> Result<MacroBody, PushError> {
+        // If the accounts tree is not complete nothing can be done
+        if !self.accounts_complete() {
+            // Maybe introduce a proper error here. Is this even needed?
```

Comment:
Correct. Deeming a proposal invalid in the case where you cannot verify its validity is the only safe option. I do not quite know about wether or not a validator with an incomplete tree can consider itself active, as i have not worked on it. I would assume, it is not fit for validating with an incomplete tree, but i would have to check with who worked on that.
This might thus be `unreachable!()`.